frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -48,22 +54,65 @@ export const surveysLogic = kea<surveysLogicType>([
         setSearchTerm: (searchTerm: string) => ({ searchTerm }),
         setSurveysFilters: (filters: Partial<SurveysFilters>, replace?: boolean) => ({ filters, replace }),
         setTab: (tab: SurveysTabs) => ({ tab }),
+        loadNextPage: true,
+        loadBackendSearchResults: true,
     }),
     loaders(({ values }) => ({
         surveys: {
-            __default: [] as Survey[],
+            __default: { results: [], count: 0 } as CountedPaginatedResponse<Survey>,
             loadSurveys: async () => {
-                const responseSurveys = await api.surveys.list()
-                return responseSurveys.results
+                return await api.surveys.list()
+            },
+            loadNextPage: async () => {
+                if (!values.surveys.next || !isURL(values.surveys.next)) {
+                    return values.surveys
+                }
+
+                const url = new URL(values.surveys.next)
+                const limit = parseInt(url.searchParams.get('limit') || SURVEY_PAGE_SIZE.toString())
+                const offset = parseInt(url.searchParams.get('offset') || values.surveys.results.length.toString())
+
+                const response = await api.surveys.list({ limit, offset })
+
+                // deduplicate results
+                const existingIds = new Set(values.surveys.results.map((s) => s.id))
+                const newResults = response.results.filter((s) => !existingIds.has(s.id))
+
+                return {
+                    ...response,
+                    results: [...values.surveys.results, ...newResults],
+                }
+            },
+            loadBackendSearchResults: async () => {
+                if (!values.searchTerm || !hasNextPage(values.surveys)) {
+                    return values.surveys
+                }
+
+                const response = await api.surveys.list({
+                    search: values.searchTerm,
+                })
+
+                const existingIds = new Set(values.surveys.results.map((s) => s.id))
+                const newResults = response.results.filter((s) => !existingIds.has(s.id))
+
+                return {
+                    ...values.surveys,
+                    results: [...values.surveys.results, ...newResults],
+                }
             },
             deleteSurvey: async (id) => {
                 await api.surveys.delete(id)
-                return values.surveys.filter((survey) => survey.id !== id)
+                return {
+                    ...values.surveys,
+                    results: values.surveys.results.filter((survey) => survey.id !== id),
+                }
```

Comment:
not sure if we should do this since I don't know what `api.surveys.delete` does exactly but surveys are never deleted, its just a logical exclusion, setting archievd=true in the database, so the survey should still be listed but its marked as archived.

Reply:
actually archived is different than delete. Archive is only available for non-draft surveys, and this one actually deletes it. the other one is a `PATCH` request that only sets the archive=true

Reply:
ah ok figured out, you can only archive if the survey is 'completed', thanks